src_c/surface.c

```diff
@@ -1830,6 +1834,16 @@ surf_convert_alpha(PyObject *self, PyObject *args)
     if (!PyArg_ParseTuple(args, "|O!", &pgSurface_Type, &srcsurf))
         return NULL;
 
+#if IS_SDLv2
```

Comment:
I think pygame SDL1 also needs set_mode to be called first for this, and raises an exception.

surf.convert() probably also needs this check